### PR TITLE
Added international language - Chinese Simplified (zh-CN)

### PR DESCRIPTION
**Pull request by ittat**

### CODE REVIEW

**Comment by fabricedesre** [comment]
Hi @ittat thanks for doing that!

I don't read Chinese so I'll ask a friend to take a look :)

However, can you split out the keyboard change to its own PR? Also, is that new code or is it reused from another project? If so, we need to add the appropriate license etc.

**Comment by ittat** [comment]
Thanks for the review, yes, the keyboard code is not part of this PR, I've removed it

**Comment by fabricedesre** [comment]
@seinlin when you have time do you mind taking a look?

**Comment by seinlin** [comment]
Sure, I will finish it today.

File: apps/ftu/locales/zh-CN/main.ftl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+wifi-password-desc =
+  .label = 输入网络密码：
+
+# 日期和时间面板
+datetime-title = 日期和时间
+datetime-clock-automatic-update = 自动更新时间
+datetime-clock-desc =
+  .label = 输入日期和时间
+datetime-clock-set = 设置日期和时间
+datetime-clock-change-success = 时间和日期更改成功
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:27]: s/时间和日期更改成功/更改时间和日期成功/

File: apps/ftu/locales/zh-CN/main.ftl [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+datetime-clock-change-error = 更改时间和日期失败
+datetime-timezone-automatic-update = 自动更新时区
+datetime-timezone-change-success = 时区更改成功
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:30]: s/时区更改成功/更改时区成功/

File: apps/ftu/locales/zh-CN/main.ftl [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+datetime-timezone-change-error = 更改时区失败
+
+# 隐私面板
+privacy-title = 隐私
+privacy-geolocation-enabled = 启用地理定位
+privacy-tor-enabled = 启用Tor
+privacy-tp-enabled = 启用跟踪保护
+
+# 显示首选项面板
+display-title = 显示首选项
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:40]: s/显示首选项/显示偏好/

File: apps/ftu/locales/zh-CN/main.ftl [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+display-prefer-dark = 使用暗模式
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:41]: s/使用暗模式/使用深色模式/

File: apps/ftu/locales/zh-CN/main.ftl [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+# 搜索引擎面板
+search-title = 搜索引擎
+search-select = 启用你的搜索引擎
+search-details = 当进行关键词搜索时，其中一个已启用的搜索引擎将被随机选择。
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:46]: s/当进行关键词搜索时，其中一个已启用的搜索引擎将被随机选择。/当进行关键词搜索时，将会随机选择一个已启用的搜索引擎。/

File: apps/ftu/locales/zh-CN/main.ftl [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+# 身份面板
+identity-title = 身份
+identity-select = 管理您的身份
+identity-details = 您可以定义尽可能多的身份以在不同的上下文中使用（例如个人与专业）。
+identity-add = 创建新身份
+identity-empty = 无可用身份
+identity-error-duplicate = 无法创建具有相同名称的多个身份！
+identity-add-input =
+  .placeholder = 新身份名称
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:63]: Looks like these two are missing. 
```
identity-device-set-name = Choose a Device Name
identity-device-desc =
  .placeholder = Device Name
```

File: apps/ftu/locales/zh-CN/main.ftl [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+webext-details = 您可以立即安装一些扩展，并在设置应用程序中管理它们。您还可以随时从Mozilla Addons网站安装扩展。
+webext-error-no-network = 请配置网络以安装扩展。
+
+# 身份面板
+identity-title = 身份
+identity-select = 管理您的身份
+identity-details = 您可以定义尽可能多的身份以在不同的上下文中使用（例如个人与专业）。
+identity-add = 创建新身份
+identity-empty = 无可用身份
+identity-error-duplicate = 无法创建具有相同名称的多个身份！
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:61]: s/无法创建具有相同名称的多个身份！/无法创用相同名称建具多个身份！/

File: apps/ftu/locales/zh-CN/main.ftl [hunk 8]
```diff
@@ -0,0 +1,63 @@
+btn-ok = 确定
+btn-cancel = 取消
+btn-back = 返回
+btn-start = 开始
+btn-done = 全部配置完成！
+
+# 语言选择面板
+language-title = 语言选择
+language-select = 选择你的语言
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:9]: s/选择你的语言/选择您的语言/

File: apps/ftu/locales/zh-CN/main.ftl [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+privacy-tor-enabled = 启用Tor
+privacy-tp-enabled = 启用跟踪保护
+
+# 显示首选项面板
+display-title = 显示首选项
+display-prefer-dark = 使用暗模式
+
+# 搜索引擎面板
+search-title = 搜索引擎
+search-select = 启用你的搜索引擎
```
Comment (seinlin) [inline apps/ftu/locales/zh-CN/main.ftl:45]: s/启用你的搜索引擎/启用您的搜索引擎/

File: apps/settings/locales/zh-CN/main.ftl [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+# Display preferences panel
+display-title = 显示偏好
+display-prefer-dark = 使用深色模式
+display-choose-homescreen = 选择主屏幕
+display-choose-theme = 选择主题
+
+# Search engines panel
+search-title = 搜索引擎
+search-select = 启用您的搜索引擎
+search-details = 当进行关键字搜索时，会随机选择一个已启用的搜索引擎。
```
Comment (seinlin) [inline apps/settings/locales/zh-CN/main.ftl:52]: s/当进行关键字搜索时，会随机选择一个已启用的搜索引擎。/当进行关键词搜索时，将会随机选择一个已启用的搜索引擎。/

File: apps/settings/locales/zh-CN/main.ftl [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+search-error-no-enabled = 请至少启用一个搜索引擎！
+
+# WebExt panel
+webext-title = Web扩展
+webext-get-more = 获取更多Web扩展
+
+# Identity panel
+identity-title = 身份
+identity-select = 管理您的身份
+identity-details = 您可以定义任意多个身份，以便在不同的上下文中使用（例如个人与职业）。
```
Comment (seinlin) [inline apps/settings/locales/zh-CN/main.ftl:62]: s/您可以定义任意多个身份，以便在不同的上下文中使用（例如个人与职业）。/您可以定义多个身份在不同的语境中使用（例如个人与专业）。/

File: apps/settings/locales/zh-CN/main.ftl [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+identity-add = 创建新身份
+identity-empty = 无可用身份
+identity-error-duplicate = 无法创建具有相同名称的多个身份！
```
Comment (seinlin) [inline apps/settings/locales/zh-CN/main.ftl:65]: s/无法创建具有相同名称的多个身份！/无法用相同名称创建多个身份！/

File: apps/settings/locales/zh-CN/main.ftl [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+identity-add-input =
+  .placeholder = 新身份名称
+
+# System Information panel
+systeminfo-title = 设备信息
+systeminfo-system-section = 系统
+systeminfo-telephony-section = 电话
+systeminfo-deviceinfo-build_number = 构建号
```
Comment (seinlin) [inline apps/settings/locales/zh-CN/main.ftl:73]: s/构建号/版本号/

File: apps/settings/locales/zh-CN/main.ftl [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+systeminfo-deviceinfo-software = 操作系统版本
+systeminfo-deviceinfo-platform_version = 平台版本
+systeminfo-deviceinfo-platform_build_id = 平台构建号
```
Comment (seinlin) [inline apps/settings/locales/zh-CN/main.ftl:76]: s/平台构建号/平台版本号/

File: apps/settings/locales/zh-CN/main.ftl [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+telephony-title = 电话
+telephony-status-enabled = 已启用
+telephony-status-disabled = 已禁用
+telephony-data-enabled = 启用移动数据
+telephony-summary = 概要
+telephony-voice = 语音
+telephony-data = 数据
+
+# Apps panel
+apps-title = 应用程序和磁贴
```
Comment (seinlin) [inline apps/settings/locales/zh-CN/main.ftl:116]: s/应用程序和磁贴/应用程序和图块/

File: apps/settings/locales/zh-CN/main.ftl [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+apps-section-title = 应用程序
+tiles-section-title = 磁贴
```
Comment (seinlin) [inline apps/settings/locales/zh-CN/main.ftl:118]: s/磁贴/图块/

File: apps/shared/resources/languages.json [hunk 17]
```diff
@@ -1,5 +1,6 @@
 {
   "en-US": "English (US)",
   "fr-FR": "Français",
-  "es-ES": "Español"
+  "es-ES": "Español",
+  "zh-CN": "中文简体"
```
Comment (seinlin) [inline apps/shared/resources/languages.json:5]: s/中文简体/简体中文/

File: apps/system/locales/zh-CN/main.ftl [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+selection-copy = 复制
+selection-share = 分享…
+selection-search = 搜索“{$query}”
+
+# 网站信息面板
+site-info-add-home = 添加到主屏幕
+site-info-split-screen = 分屏
+site-info-choose-ua = 用户代理
+site-info-b2g-ua = 默认
+site-info-android-ua = Android
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:43]: s/Android/安卓/

File: apps/system/locales/zh-CN/main.ftl [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+drm-success-title = DRM支持已成功启用
+drm-content-disabled-title = DRM支持
+drm-content-disabled-text = 您必须启用DRM才能在此页面上播放某些音频或视频
+drm-button-title = 启用
+drm-content-cdm-installing-title = DRM正在安装
+drm-content-cdm-installing-text = Capyloon正在安装播放此页面上的音频或视频所需的组件。请稍后再试。
+
+
+# P2P 发现和配对
+p2p-enable-discovery-success = 启用了对等体发现
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:142]: s/启用了对等体发现/节点发现已启用/

File: apps/system/locales/zh-CN/main.ftl [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+p2p-enable-discovery-failure = 无法启用对等体发现
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:143]: s/无法启用对等体发现/无法启用节点发现/

File: apps/system/locales/zh-CN/main.ftl [hunk 21]
```diff
@@ -0,0 +1,174 @@
...
+p2p-disable-discovery-success = 禁用了对等体发现
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:144]: s/禁用了对等体发现/节点发现已禁用/

File: apps/system/locales/zh-CN/main.ftl [hunk 22]
```diff
@@ -0,0 +1,174 @@
...
+p2p-disable-discovery-failure = 无法禁用对等体发现
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:145]: s/无法禁用对等体发现/无法禁用节点发现/

File: apps/system/locales/zh-CN/main.ftl [hunk 23]
```diff
@@ -0,0 +1,174 @@
...
+p2p-peer-found = 发现了新对等体：{$name}
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:146]: s/发现了新对等体：/发现了新节点：/

File: apps/system/locales/zh-CN/main.ftl [hunk 24]
```diff
@@ -0,0 +1,174 @@
...
+p2p-connect-error-not-connected = 无法连接 {$desc} ：未连接
+p2p-connect-error-not-paired = 无法连接 {$desc} ：未配对
+p2p-connect-error-denied = {$desc} 拒绝了连接
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:149]: s/拒绝了连接/连接已拒绝/

File: apps/system/locales/zh-CN/main.ftl [hunk 25]
```diff
@@ -0,0 +1,174 @@
...
+p2p-connect-error-other = 连接 {$desc} 时出错
+p2p-connect-success = 已连接到 {$desc}
+p2p-connect-request-title = 对等体连接请求
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:152]: s/对等体连接请求/节点连接请求/

File: apps/system/locales/zh-CN/main.ftl [hunk 26]
```diff
@@ -0,0 +1,174 @@
...
+p2p-download-reject = 取消
+p2p-launch-title = 您是否要在 {$device} 上启动由 {$source} 发送的此应用？
+p2p-launch-accept = 启动
+p2p-launch-reject = 取消
+p2p-activity-title = 您是否要在 {$device} 上启动由 {$source} 触发的此活动？
+p2p-activity-accept = 启动
+p2p-activity-reject = 取消
+p2p-tile-title = 您是否要在 {$device} 上启动由 {$source} 发送的此 Tile？
+p2p-tile-accept = 启动
+p2p-tile-reject = 取消
```
Comment (seinlin) [inline apps/system/locales/zh-CN/main.ftl:174]: nit: didn't see translation for these. 
```
# Activities names displayed in the activity chooser.
activity-pick = Choose file in…
activity-share = Share with…
activity-add-to-home = Add to Homescreen
activity-p2p-tile-start = Select Tile
activity-scan-qr-code = Scan QR Code
activity-view-resource = View File
activity-publish-resource = Publish File
activity-install-wasm-plugin = Install WASM plugin
```

**Comment by seinlin** [comment]
I finished the review. My suggest is to make it more natural and a better tone.

**Comment by ittat** [comment]
Thanks @seinlin for the review and suggestion!
I'll deal with these later

**Comment by ittat** [comment]
OK! All changes updated :)

**Comment by seinlin** [comment]
@fabricedesre, it looks good to me. @ittat  Thanks for your contribution.

**Pull request merged**